<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>3. selection — Class for analyzing a selection timeseries &mdash; Enrich2 2.0 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '2.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Enrich2 2.0 documentation" href="index.html" />
    <link rel="next" title="4. experiment — Class for comparing multiple selections" href="experiment.html" />
    <link rel="prev" title="2.6. barcodevariant — Barcoded variant sequencing library" href="barcodevariant.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="experiment.html" title="4. experiment — Class for comparing multiple selections"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="barcodevariant.html" title="2.6. barcodevariant — Barcoded variant sequencing library"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Enrich2 2.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="module-selection">
<span id="selection-class-for-analyzing-a-selection-timeseries"></span><h1>3. <a class="reference internal" href="#module-selection" title="selection: Class for analyzing a selection timeseries."><tt class="xref py py-mod docutils literal"><span class="pre">selection</span></tt></a> &#8212; Class for analyzing a selection timeseries<a class="headerlink" href="#module-selection" title="Permalink to this headline">¶</a></h1>
<p>The <a class="reference internal" href="#module-selection" title="selection: Class for analyzing a selection timeseries."><tt class="xref py py-mod docutils literal"><span class="pre">selection</span></tt></a> module contains the class definition for the <a class="reference internal" href="#selection.Selection" title="selection.Selection"><tt class="xref py py-class docutils literal"><span class="pre">Selection</span></tt></a> class, used for analyzing multiple timepoints in a selection experiment. Each timepoint consists of one or more <a class="reference external" href="seqlib_overview.html">Seqlib family objects</a>.</p>
<div class="section" id="configuration-options">
<h2>3.1. Configuration options<a class="headerlink" href="#configuration-options" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt><strong>&#8216;name&#8217;</strong>  - <em>required</em></dt>
<dd>The name of this entity. This used for output directory names, and is included in every error message and log message. Names should be unique within a single analysis, but this is not enforced.</dd>
<dt><strong>&#8216;output directory&#8217;</strong></dt>
<dd><p class="first">Sets the base output directory for the analysis. Output files and directories will be created in subdirectories at this location. It will be created if it doesn&#8217;t exist.</p>
<p>If desired, the output directory can be set for each entity separately. If no &#8216;output directory&#8217; is included in the config, the entity will use the parent&#8217;s setting.</p>
<p class="last">This setting is required for the top level entity (usually an <a class="reference internal" href="experiment.html#experiment.Experiment" title="experiment.Experiment"><tt class="xref py py-class docutils literal"><span class="pre">Experiment</span></tt></a>) in each analysis.</p>
</dd>
<dt><strong>&#8216;libraries&#8217;</strong> - <em>required</em></dt>
<dd>List of <a class="reference internal" href="seqlib.html#seqlib.seqlib.SeqLib" title="seqlib.seqlib.SeqLib"><tt class="xref py py-class docutils literal"><span class="pre">SeqLib</span></tt></a> config objects for sequencing libraries that are part of this <a class="reference internal" href="#selection.Selection" title="selection.Selection"><tt class="xref py py-class docutils literal"><span class="pre">Selection</span></tt></a>.</dd>
<dt><strong>&#8216;barcodes&#8217;</strong></dt>
<dd><p class="first">Barcode options.</p>
<dl class="last docutils">
<dt><strong>&#8216;map file&#8217;</strong></dt>
<dd>Path to the <a class="reference internal" href="barcodevariant.html#seqlib.barcodevariant.BarcodeMap" title="seqlib.barcodevariant.BarcodeMap"><tt class="xref py py-class docutils literal"><span class="pre">BarcodeMap</span></tt></a> file mapping barcodes to variant sequences. This <a class="reference internal" href="barcodevariant.html#seqlib.barcodevariant.BarcodeMap" title="seqlib.barcodevariant.BarcodeMap"><tt class="xref py py-class docutils literal"><span class="pre">BarcodeMap</span></tt></a> will be used for any <a class="reference internal" href="barcodevariant.html#seqlib.barcodevariant.BarcodeVariantSeqLib" title="seqlib.barcodevariant.BarcodeVariantSeqLib"><tt class="xref py py-class docutils literal"><span class="pre">BarcodeVariantSeqLib</span></tt></a> libraries that do not have their own specified.</dd>
</dl>
</dd>
<dt><strong>&#8216;filters&#8217;</strong> - <em>required</em></dt>
<dd><p class="first">Filtering options for barcodes and variants.</p>
<dl class="last docutils">
<dt><strong>&#8216;min count&#8217;</strong></dt>
<dd>Minimum count for this barcode or variant at any timepoint.</dd>
<dt><strong>&#8216;min input count&#8217;</strong></dt>
<dd>Minimum count for this barcode or variant in the input (timepoint 0).</dd>
<dt><strong>&#8216;min rsquared&#8217;</strong></dt>
<dd>Minimum r-squared value for scored variants and barcodes.</dd>
<dt><strong>&#8216;max barcode variation&#8217;</strong></dt>
<dd>Maximum coefficient of variation for barcode scores mapped to the same variant. This is only applied when only when all sequencing data has both barcode and variant data (<em>i.e.</em> are <a class="reference internal" href="barcodevariant.html#seqlib.barcodevariant.BarcodeVariantSeqLib" title="seqlib.barcodevariant.BarcodeVariantSeqLib"><tt class="xref py py-class docutils literal"><span class="pre">BarcodeVariantSeqLib</span></tt></a> objects) with the same <a class="reference internal" href="barcodevariant.html#seqlib.barcodevariant.BarcodeMap" title="seqlib.barcodevariant.BarcodeMap"><tt class="xref py py-class docutils literal"><span class="pre">BarcodeMap</span></tt></a>.</dd>
</dl>
</dd>
<dt><strong>&#8216;carryover correction&#8217;</strong></dt>
<dd><p class="first">Nonspecific carryover correction options.</p>
<dl class="last docutils">
<dt><strong>&#8216;method&#8217;</strong></dt>
<dd>Nonspecific carryover correction method. Currently, only &#8220;nonsense&#8221; is supported (described in <a class="reference external" href="http://www.pnas.org/content/109/42/16858.abstract">Araya and Fowler</a>).</dd>
<dt><strong>&#8216;position&#8217;</strong></dt>
<dd>Additional parameterd for &#8220;nonsense&#8221; method. This is the last amino acid position (first position is 1) for which a change to stop marks a variant as part of nonspecific carryover.</dd>
</dl>
</dd>
<dt><strong>&#8216;normalize wt&#8217;</strong></dt>
<dd>If this is <tt class="docutils literal"><span class="pre">True</span></tt>, normalize all scores or ratios using the wild type score or ratio as neutral.</dd>
</dl>
</div>
<div class="section" id="selection-class">
<h2>3.2. <a class="reference internal" href="#selection.Selection" title="selection.Selection"><tt class="xref py py-class docutils literal"><span class="pre">Selection</span></tt></a> class<a class="headerlink" href="#selection-class" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="selection.Selection">
<em class="property">class </em><tt class="descclassname">selection.</tt><tt class="descname">Selection</tt><big>(</big><em>config</em><big>)</big><a class="headerlink" href="#selection.Selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Class for a single selection experiment, consisting of multiple 
timepoints. This class coordinates <a class="reference internal" href="seqlib.html#seqlib.seqlib.SeqLib" title="seqlib.seqlib.SeqLib"><tt class="xref py py-class docutils literal"><span class="pre">SeqLib</span></tt></a> 
objects. Creating a <a class="reference internal" href="#selection.Selection" title="selection.Selection"><tt class="xref py py-class docutils literal"><span class="pre">Selection</span></tt></a> requires a valid 
<em>config</em> object, usually from a <tt class="docutils literal"><span class="pre">.json</span></tt> configuration file.</p>
<dl class="method">
<dt id="selection.Selection.add_variants_to_barcodes">
<tt class="descname">add_variants_to_barcodes</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.add_variants_to_barcodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Add the associated variant information to each row of the barcode <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a>.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.calc_barcode_variation">
<tt class="descname">calc_barcode_variation</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.calc_barcode_variation" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the <a class="reference external" href="http://en.wikipedia.org/wiki/Coefficient_of_variation">coefficient of variation</a> 
for each variant&#8217;s barcode enrichment scores. Requires both variant and barcode 
data for all timepoints.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.calc_counts">
<tt class="descname">calc_counts</tt><big>(</big><em>dtype</em><big>)</big><a class="headerlink" href="#selection.Selection.calc_counts" title="Permalink to this definition">¶</a></dt>
<dd><p>Tabulate counts for each timepoint and create the <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> indicated by 
<em>dtype</em>. All <a class="reference internal" href="seqlib.html#seqlib.seqlib.SeqLib" title="seqlib.seqlib.SeqLib"><tt class="xref py py-class docutils literal"><span class="pre">SeqLib</span></tt></a> objects need to be counted before calling 
this method.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.calc_enrichments">
<tt class="descname">calc_enrichments</tt><big>(</big><em>dtype</em><big>)</big><a class="headerlink" href="#selection.Selection.calc_enrichments" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate enrichment scores and r-squared values for each element in the <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> indicated by 
<em>dtype</em> (&#8216;variant&#8217; or &#8216;barcode&#8217;). Assumes ratios have been 
calculated by <a class="reference internal" href="#selection.Selection.calc_ratios" title="selection.Selection.calc_ratios"><tt class="xref py py-meth docutils literal"><span class="pre">calc_ratios()</span></tt></a>. Calculations performed using 
<tt class="xref py py-func docutils literal"><span class="pre">linear_enrichment_apply_fn()</span></tt>.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.calc_frequencies">
<tt class="descname">calc_frequencies</tt><big>(</big><em>dtype</em><big>)</big><a class="headerlink" href="#selection.Selection.calc_frequencies" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate frequencies for each element in the <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> indicated by 
<em>dtype</em> (&#8216;variant&#8217; or &#8216;barcode&#8217;).</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.calc_ratios">
<tt class="descname">calc_ratios</tt><big>(</big><em>dtype</em><big>)</big><a class="headerlink" href="#selection.Selection.calc_ratios" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate ratios for each element in the <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> indicated by 
<em>dtype</em> (&#8216;variant&#8217; or &#8216;barcode&#8217;). Assumes frequencies have been 
calculated by <a class="reference internal" href="#selection.Selection.calc_frequencies" title="selection.Selection.calc_frequencies"><tt class="xref py py-meth docutils literal"><span class="pre">calc_frequencies()</span></tt></a>.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.calculate">
<tt class="descname">calculate</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.calculate" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper method to calculate counts, frequencies, ratios, and enrichment scores 
for all data in the <a class="reference internal" href="#selection.Selection" title="selection.Selection"><tt class="xref py py-class docutils literal"><span class="pre">Selection</span></tt></a>. Enrichment scores are only calculated 
if there are more than two timepoints.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.count_mutations">
<tt class="descname">count_mutations</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.count_mutations" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates and populates <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> objects for individual mutations. This method 
should be called after all filtering has been completed. The new 
<a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> objects have dtype &#8216;mutations_nt&#8217; and &#8216;mutations_aa&#8217; (only if the 
data set is coding).</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.count_timepoints">
<tt class="descname">count_timepoints</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.count_timepoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Combine <a class="reference internal" href="seqlib.html#seqlib.seqlib.SeqLib" title="seqlib.seqlib.SeqLib"><tt class="xref py py-class docutils literal"><span class="pre">SeqLib</span></tt></a> objects into individual timepoints and 
tabulate counts for each timepoint. Counts are stored in the local
<a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a>. To tabulate counts for individual mutations 
(not variants), see <a class="reference internal" href="#selection.Selection.count_mutations" title="selection.Selection.count_mutations"><tt class="xref py py-meth docutils literal"><span class="pre">count_mutations()</span></tt></a>.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.filter_variant_data">
<tt class="descname">filter_variant_data</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.filter_variant_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply the filtering functions to the variant data, based on the filter 
options present in the configuration object. Filtering is performed 
using the appropriate apply function. Frequencies, ratios, and 
enrichments must be recalculated after filtering.</p>
<p>The data are written to the subdirectory <tt class="docutils literal"><span class="pre">&quot;pre-filter&quot;</span></tt> before filtering.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.library_list">
<tt class="descname">library_list</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.library_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the <a class="reference internal" href="seqlib.html#seqlib.seqlib.SeqLib" title="seqlib.seqlib.SeqLib"><tt class="xref py py-class docutils literal"><span class="pre">SeqLib</span></tt></a> objects as a list, sorted by timepoint.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.nonspecific_carryover">
<tt class="descname">nonspecific_carryover</tt><big>(</big><em>ns_apply_fn</em>, <em>**kwargs</em><big>)</big><a class="headerlink" href="#selection.Selection.nonspecific_carryover" title="Permalink to this definition">¶</a></dt>
<dd><p>Correct the counts in the &#8216;variants&#8217; <a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v0.13.1)"><tt class="xref py py-class docutils literal"><span class="pre">pandas.DataFrame</span></tt></a> for nonspecific carryover. 
Nonspecific counts are defined by <em>ns_apply_fn</em> and its <em>kwargs</em>, which 
takes a row as an argument and returns <tt class="docutils literal"><span class="pre">True</span></tt> if the row&#8217;s counts 
are nonspecific.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.normalize_variants_to_wt">
<tt class="descname">normalize_variants_to_wt</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.normalize_variants_to_wt" title="Permalink to this definition">¶</a></dt>
<dd><p>Normalizes variant scores (or ratios if only two timepoints are present) such that the wild type value is &#8220;neutral&#8221; (0 for scores, 1 for ratios). Does nothing for barcode-only data.</p>
<p>The data are written to the subdirectory <tt class="docutils literal"><span class="pre">&quot;pre-wtnorm&quot;</span></tt> before normalization.</p>
</dd></dl>

<dl class="method">
<dt id="selection.Selection.use_scores">
<tt class="descname">use_scores</tt><big>(</big><big>)</big><a class="headerlink" href="#selection.Selection.use_scores" title="Permalink to this definition">¶</a></dt>
<dd><p>Return <tt class="docutils literal"><span class="pre">True</span></tt> if scores (slopes) are used, else <tt class="docutils literal"><span class="pre">False</span></tt> (use ratios).</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="selection-apply-functions">
<h2>3.3. <a class="reference internal" href="#module-selection" title="selection: Class for analyzing a selection timeseries."><tt class="xref py py-mod docutils literal"><span class="pre">selection</span></tt></a> apply functions<a class="headerlink" href="#selection-apply-functions" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="selection.nonsense_ns_carryover_apply_fn">
<tt class="descclassname">selection.</tt><tt class="descname">nonsense_ns_carryover_apply_fn</tt><big>(</big><em>row</em>, <em>position</em><big>)</big><a class="headerlink" href="#selection.nonsense_ns_carryover_apply_fn" title="Permalink to this definition">¶</a></dt>
<dd><p><a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.apply.html#pandas.DataFrame.apply" title="(in pandas v0.13.1)"><tt class="xref py py-meth docutils literal"><span class="pre">pandas.DataFrame.apply()</span></tt></a> function for determining which rows 
contribute counts to nonspecific carryover calculations. Returns <tt class="docutils literal"><span class="pre">True</span></tt> 
if the variant has a change to stop at or before amino acid number 
<em>position</em>.</p>
</dd></dl>

<div class="section" id="filtering-apply-functions">
<h3>3.3.1. Filtering apply functions<a class="headerlink" href="#filtering-apply-functions" title="Permalink to this headline">¶</a></h3>
<dl class="function">
<dt id="selection.min_count_filter">
<tt class="descclassname">selection.</tt><tt class="descname">min_count_filter</tt><big>(</big><em>row</em>, <em>cutoff</em><big>)</big><a class="headerlink" href="#selection.min_count_filter" title="Permalink to this definition">¶</a></dt>
<dd><p>Filtering function for minimum counts across all timepoints.</p>
</dd></dl>

<dl class="function">
<dt id="selection.min_input_count_filter">
<tt class="descclassname">selection.</tt><tt class="descname">min_input_count_filter</tt><big>(</big><em>row</em>, <em>cutoff</em><big>)</big><a class="headerlink" href="#selection.min_input_count_filter" title="Permalink to this definition">¶</a></dt>
<dd><p>Filtering function for minimum count in input timepoint.</p>
</dd></dl>

<dl class="function">
<dt id="selection.min_rsq_filter">
<tt class="descclassname">selection.</tt><tt class="descname">min_rsq_filter</tt><big>(</big><em>row</em>, <em>cutoff</em><big>)</big><a class="headerlink" href="#selection.min_rsq_filter" title="Permalink to this definition">¶</a></dt>
<dd><p>Filtering function for minimum r-squared value. Entries with no 
r-squared value are retained.</p>
</dd></dl>

</div>
<div class="section" id="barcodevariant-specific-apply-functions">
<h3>3.3.2. <tt class="xref py py-class docutils literal"><span class="pre">BarcodeVariant</span></tt>-specific apply functions<a class="headerlink" href="#barcodevariant-specific-apply-functions" title="Permalink to this headline">¶</a></h3>
<dl class="function">
<dt id="selection.barcode_variation_apply_fn">
<tt class="descclassname">selection.</tt><tt class="descname">barcode_variation_apply_fn</tt><big>(</big><em>row</em>, <em>barcode_data</em>, <em>mapping</em><big>)</big><a class="headerlink" href="#selection.barcode_variation_apply_fn" title="Permalink to this definition">¶</a></dt>
<dd><p><a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.apply.html#pandas.DataFrame.apply" title="(in pandas v0.13.1)"><tt class="xref py py-meth docutils literal"><span class="pre">pandas.DataFrame.apply()</span></tt></a> function for calculating the coefficient 
of variation for a variant&#8217;s barcodes.</p>
</dd></dl>

<dl class="function">
<dt id="selection.barcode_count_apply_fn">
<tt class="descclassname">selection.</tt><tt class="descname">barcode_count_apply_fn</tt><big>(</big><em>row</em>, <em>mapping</em><big>)</big><a class="headerlink" href="#selection.barcode_count_apply_fn" title="Permalink to this definition">¶</a></dt>
<dd><p><a class="reference external" href="http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.apply.html#pandas.DataFrame.apply" title="(in pandas v0.13.1)"><tt class="xref py py-meth docutils literal"><span class="pre">pandas.DataFrame.apply()</span></tt></a> function for counting the number of 
unique barcodes for a variant.</p>
</dd></dl>

<dl class="function">
<dt id="selection.barcode_varation_filter">
<tt class="descclassname">selection.</tt><tt class="descname">barcode_varation_filter</tt><big>(</big><em>row</em>, <em>cutoff</em><big>)</big><a class="headerlink" href="#selection.barcode_varation_filter" title="Permalink to this definition">¶</a></dt>
<dd><p>Filtering function for barcode coefficient of variation.</p>
</dd></dl>

</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">3. <tt class="docutils literal"><span class="pre">selection</span></tt> &#8212; Class for analyzing a selection timeseries</a><ul>
<li><a class="reference internal" href="#configuration-options">3.1. Configuration options</a></li>
<li><a class="reference internal" href="#selection-class">3.2. <tt class="docutils literal"><span class="pre">Selection</span></tt> class</a></li>
<li><a class="reference internal" href="#selection-apply-functions">3.3. <tt class="docutils literal"><span class="pre">selection</span></tt> apply functions</a><ul>
<li><a class="reference internal" href="#filtering-apply-functions">3.3.1. Filtering apply functions</a></li>
<li><a class="reference internal" href="#barcodevariant-specific-apply-functions">3.3.2. <tt class="docutils literal"><span class="pre">BarcodeVariant</span></tt>-specific apply functions</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="barcodevariant.html"
                        title="previous chapter">2.6. <tt class="docutils literal docutils literal docutils literal docutils literal"><span class="pre">barcodevariant</span></tt> &#8212; Barcoded variant sequencing library</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="experiment.html"
                        title="next chapter">4. <tt class="docutils literal docutils literal docutils literal docutils literal"><span class="pre">experiment</span></tt> &#8212; Class for comparing multiple selections</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/selection.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="experiment.html" title="4. experiment — Class for comparing multiple selections"
             >next</a> |</li>
        <li class="right" >
          <a href="barcodevariant.html" title="2.6. barcodevariant — Barcoded variant sequencing library"
             >previous</a> |</li>
        <li><a href="index.html">Enrich2 2.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Alan F Rubin and Douglas M Fowler.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.1.
    </div>
  </body>
</html>